src/table/body-cell/inline-editor.tsx

```diff
@@ -112,7 +111,7 @@ export function InlineEditor<ItemType>({
         aria-label={ariaLabels?.activateEditLabel?.(column, item)}
         onKeyDown={handleEscape}
       >
-        <form onSubmit={onSubmitClick} className={styles['body-cell-editor-form']}>
+        <div className={styles['body-cell-editor-form']}>
```

Comment:
I get the motivation behind this change. Out of curiosity: would the change from a `from` to a `div` element impact a11y?  Semantically a form would be correct because the tag contains interactive controls for submitting information ([MDN reference](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form)).

Reply:
I don't think the form is needed there for accessibility but that needs checking. The editor is wrapped with a dialog role which makes it similar to some other components of ours, for example collection preferences or split panel settings. We do not use the `form` in either.

Reply:
Oh, and the date range picker as well

Reply:
This breaks auto submitting edits on "Enter" feature. 

I would recommend finding alternative fixes for the issue. Form tag should stay, because it is semantically expected here

Reply:
Yes, that is true. I updated my implementation.

Reply:
This version does not support custom controls, as it introduces secret props unavailable for consumers.

There must be a simple solution and there is. Talked to the original customer, recommended to abandon inline editing feature all together. 

They do not depend on this PR anymore, so we can close it and claim this use-case unsupported